gtsam/base/Lie.h

```diff
@@ -319,11 +320,33 @@ T expm(const Vector& x, int K=7) {
 }
 
 /**
- * Linear interpolation between X and Y by coefficient t in [0, 1].
+ * Linear interpolation between X and Y by coefficient t in [0, 1.5] (t>1 implies extrapolation), with optional jacobians.
  */
 template <typename T>
-T interpolate(const T& X, const T& Y, double t) {
-  assert(t >= 0 && t <= 1);
+T interpolate(const T& X, const T& Y, double t,
+              OptionalJacobian< traits<T>::dimension, traits<T>::dimension > Hx = boost::none,
+              OptionalJacobian< traits<T>::dimension, traits<T>::dimension > Hy = boost::none) {
+  assert(t >= 0.0 && t <= 1.5);
+  if (Hx || Hy) {
+    typedef Eigen::Matrix<double, traits<T>::dimension, traits<T>::dimension> Jacobian;
+    typename traits<T>::TangentVector log_Xinv_Y;
+    Jacobian Hx_tmp, Hy_tmp, H1, H2;
```

Comment:
How about giving these informative names e.g. `XinvY_H_x` and do a simple multiplication in lines 346 and 347. It might also be interesting to look at the generic implementations of compose and between:
```
  Class compose(const Class& g, ChartJacobian H1,
      ChartJacobian H2 = boost::none) const {
    if (H1) *H1 = g.inverse().AdjointMap();
    if (H2) *H2 = Eigen::Matrix<double, N, N>::Identity();
    return derived() * g;
  }

  Class between(const Class& g, ChartJacobian H1,
      ChartJacobian H2 = boost::none) const {
    Class result = derived().inverse() * g;
    if (H1) *H1 = - result.inverse().AdjointMap();
    if (H2) *H2 = Eigen::Matrix<double, N, N>::Identity();
    return result;
  }
```
from which you can see the `H2` in both cases is identity, allowing some simplification below as well.

Reply:
great suggestion - done